### PR TITLE
MAGN-8734 Startup crash with Dynamo/Revit2016 points to SharpDX

### PR DESCRIPTION
**Pull request by ikeough**
### Purpose

This PR continues work on [MAGN-8734](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8734). 

- Through investigation with @mjkkirschner, it was found that the SharpDX package requires and environment variable to ensure that the correct dlls are being pulled from the SharpDX package folder during build. This environment variable is now set in our `CS.props` file.
- It was also found that two shader compilation flags, added during the switch-over to using the HelixToolkit packages, caused all shaders compiled using these flags to fail on Parallels. 
- The inclusion of the SharpDX packages also caused duplicate versions of SharpDX dlls to be put into the nodes folder. The `DynamoWpf.targets` file has been updated to remove all of these in a post build step to avoid the failure to resolve methods at run-time due to conflicting versions of dlls.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
  - Uneccessary dlls are removed from the output. 
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
  - No new documentation added. This is a configuration only change.
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
  - No new user-facing strings are added.

### Reviewers

@mjkkirschner 

### FYIs

@jnealb, @lillismith, @Racel

### CODE REVIEW

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj [hunk 1]
```diff
@@ -1258,13 +1256,10 @@
...
     </Content>
-    <Content Include="d3dcompiler_44.dll">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </Content>
-    <Content Include="sharpdx_direct3d11_1_effects_x64.dll">
+    <Content Include="sharpdx_direct3d11_effects_x64.dll">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </Content>
-    <Content Include="sharpdx_direct3d11_effects_x64.dll">
+    <Content Include="sharpdx_direct3d11_effects_x86.dll">
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj]: is this really what we want here? I thought we wanted x64

File: src/DynamoCoreWpf/ViewModels/Watch3D/Shaders/CompileShaders.bat [hunk 2]
```diff
@@ -1,6 +1,6 @@
 @echo off
 call "%DXSDK_DIR%Utilities\bin\dx_setenv.cmd"
 pushd %~dp0
-fxc /Od /Zi /T fx_5_0 /I ./../ /Fo ..\Resources\_dynamo.bfx Default.fx
+fxc /T fx_5_0 /I ./../ /Fo ..\Resources\_dynamo.bfx Default.fx
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Watch3D/Shaders/CompileShaders.bat]: !!!

File: src/Libraries/DynamoWatch3D/Watch3D.csproj [hunk 3]
```diff
@@ -44,19 +44,19 @@
   <ItemGroup>
     <Reference Include="Cyotek.Drawing.BitmapFont, Version=1.0.2.0, Culture=neutral, PublicKeyToken=58daa28b0b2de221, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Cyotek.Drawing.BitmapFont.1.0.2.0\lib\net20\Cyotek.Drawing.BitmapFont.dll</HintPath>
-      <Private>False</Private>
+      <Private>True</Private>
```
Comment (mjkkirschner) [inline src/Libraries/DynamoWatch3D/Watch3D.csproj]: should these be true? I think this is somehow related to copy local...

**Comment by mjkkirschner** [comment]
this branch has some merge conflicts, is this the 2016 branch trying to be merged to 2015 by chance? or maybe some issues with strangely formatted project files?

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj (hunk 1 above)
Comment (ikeough) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj]: I think this is just some merge weirdness related to packages. Both versions of that dll are definitely still in the project and both are set to `Copy Always`

File: src/Libraries/DynamoWatch3D/Watch3D.csproj (hunk 3 above)
Comment (ikeough) [inline src/Libraries/DynamoWatch3D/Watch3D.csproj]: Same as above, when you bring in packages, the dlls get set to copy local. I've got a cleanup target that removes extraneous SharpDX dlls from the nodes folder.

**Comment by Racel** [comment]
@mjkkirschner - Is there an update on this? Just curious as I am still unable to load sandbox.

**Comment by mjkkirschner** [comment]
@Racel yes we found that you can enable hardware acceleration in parallels but we needed to recompile some shaders with optimizations on and debug info off.

**Comment by ikeough** [comment]
@mjkkirschner I've resolved the merge conflicts. Please take a look again.

**Comment by mjkkirschner** [comment]
@ikeough I think looks good, but I wanted to clarify that while parallels with hardware acceleration on- now works great!! (yay!)

parallels with hardware acceleration turned off, does not seem to fallback to WARP gracefully, though there is no more crash:

screen shot 2015-10-05 at 1 25 21 pm

**Comment by jnealb** [comment]
@ikeough @mjkkirschner Wow! Nice going.

**Comment by Racel** [comment]
@mjkkirschner - I am getting the same error with hardware acceleration turned off. Are you going to file a new task?

**Pull request merged**